bin/s2nc.c

```diff
@@ -447,6 +466,20 @@ int main(int argc, char *const *argv)
         case OPT_NPN:
             npn = true;
             break;
+        case OPT_BUFFERED_SEND:
+            send_buffer_size_scanned_value = strtol(optarg, 0, 10);
+            if (send_buffer_size_scanned_value > UINT32_MAX || send_buffer_size_scanned_value < 0) {
+                fprintf(stderr, "<buffer size> must be a positive 32 bit value\n");
+                exit(1);
+            }
+            send_buffer_size = (uint32_t) send_buffer_size_scanned_value;
+            break;
```

Comment:
The scope of send_buffer_size_scanned_value threw me a bit. You can reduce the scope (and the need to the very specific long variable name) with curly braces in your case statement:
```suggestion
        case OPT_BUFFERED_SEND: {
            long arg = strtol(optarg, 0, 10);
            if (arg > UINT32_MAX || arg < 0) {
                fprintf(stderr, "<buffer size> must be a positive 32 bit value\n");
                exit(1);
            }
            send_buffer_size = (uint32_t) arg;
            break;
        }
```